test/gtest/median.hpp

```diff
@@ -0,0 +1,333 @@
+        {
+            output_dims.erase(output_dims.begin() + dim);
+            if(output_dims.empty())
+                output_dims.push_back(1);
+        }
+        else
+        {
+            output_dims[dim] = 1;
+        }
+
+        auto in_gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+        input             = tensor<T>{input_dims, input_strides}.generate(in_gen_value);
+
+        output = tensor<T>{output_dims};
+        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        indices = tensor<size_t>{output_dims};
+        std::fill(indices.begin(), indices.end(), 0);
+
+        ref_output = tensor<T>{output_dims};
+        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+
+        ref_indices = tensor<size_t>{output_dims};
+        std::fill(ref_indices.begin(), ref_indices.end(), 0);
+
+        input_dev   = handle.Write(input.data);
+        output_dev  = handle.Write(output.data);
+        indices_dev = handle.Write(indices.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        miopenStatus_t status;
+
+        // Run cpu
+        cpu_median_fwd(input, ref_output, ref_indices, dim);
+
+        // Run kernel
+        status = miopen::median::MedianForward(handle,
+                                               input.desc,
+                                               input_dev.get(),
+                                               output.desc,
+                                               output_dev.get(),
+                                               indices.desc,
+                                               (size_t*)indices_dev.get(),
+                                               dim,
+                                               keepdim);
+
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        // Copy output data from device to host
+        output.data  = handle.Read<T>(output_dev, output.data.size());
+        indices.data = handle.Read<size_t>(indices_dev, indices.data.size());
+    }
+
+    double GetTolerance()
+    {
+        double tolerance = std::numeric_limits<T>::epsilon() * 10;
+        return tolerance;
+    }
+
+    void Verify()
+    {
+        // Verify output_tensor
+        double threshold = GetTolerance();
+        auto error       = miopen::rms_range(ref_output, output);
+        EXPECT_LT(error, threshold) << "Error output beyond tolerance Error: " << error
+                                    << ", Threshold: " << threshold << std::endl;
+
+        // Verify indices_tensor
+        ASSERT_TRUE(miopen::range_distance(ref_indices) == miopen::range_distance(indices));
+        for(size_t i = 0; i < indices.data.size(); i++)
+        {
+            if(indices.data[i] != ref_indices.data[i])
+            {
+                ASSERT_TRUE(output.data[i] == output.data[i]) << "Error output (indices) mismatch";
+            }
+        }
```

Comment:
Why don't you use `EXPECT_LT(error, threshold)` like what you did with the output.

Reply:
for `indices`, the dtype is integer, hence, it's expected to be exactly the same for `output` and `ref_output` instead of checking for a `threshold` like floating-point types

Reply:
This loop was initially designed to verify cases where multiple valid medians exist, resulting in multiple valid indices. However, the original logic I implemented was incorrect. I’ve since fixed it—please take a look:
```
// Verify indices_tensor
ASSERT_EQ(miopen::range_distance(ref_indices), miopen::range_distance(indices));
auto input_tv      = miopen::get_inner_expanded_tv<5>(input.desc);
auto indices_numel = indices.desc.GetElementSize();
auto reduce_size   = input.desc.GetLengths()[dim];
auto inner_size    = std::accumulate(input.desc.GetLengths().begin() + dim + 1,
                                  input.desc.GetLengths().end(),
                                  1ULL,
                                  std::multiplies<size_t>());

for(auto i = 0; i < indices_numel; ++i)
{
    auto local_idx     = indices.data[i];
    auto ref_local_idx = ref_indices.data[i];

    if(local_idx != ref_local_idx)
    {
        auto idx = (i / inner_size) * inner_size * reduce_size + i % inner_size +
                   local_idx * inner_size;
        auto ref_idx = (i / inner_size) * inner_size * reduce_size + i % inner_size +
                       ref_local_idx * inner_size;

        tensor_layout_t<5> input_layout(input_tv, idx);
        tensor_layout_t<5> ref_input_layout(input_tv, ref_idx);

        auto global_idx     = input_tv.get_tensor_view_idx(input_layout);
        auto ref_global_idx = input_tv.get_tensor_view_idx(ref_input_layout);

        ASSERT_EQ(input.data[global_idx], input.data[ref_global_idx])
            << "Error output (indices) mismatch." << std::endl;
    }
}
```

Reply:
There is something wrong with your new implementation